tools/codesigningtool/codesigningtool.py

```diff
@@ -78,17 +78,26 @@ def _get_identities_from_provisioning_profile(mpf):
         yield _certificate_fingerprint(identity.data)
 
 
-def _find_codesign_identities():
+def _find_codesign_identities(identity=None):
     """Finds code signing identities on the current system."""
     ids = []
     output, stderr = _check_output([
         "security", "find-identity", "-v", "-p", "codesigning",
     ])
     output = output.decode("utf-8").strip()
+    pattern = "(?P<hash>[A-F0-9]{40})"
+    if identity:
+        pattern += r'\s+"(?P<full_name>.*?{}.*?)"'.format(identity)
```

Comment:
fyi - we'll tweak this as it lands, I think this should call `re.escape` on `identity` before inserting it into the regex string.  Since it comes from the developers account, it likely could have characters that would otherwise having meaning in a regex.